### PR TITLE
fix(gisaid-upload) Change validator structure for GisaidSample

### PR DESCRIPTION
**Pull request by Vince-janv**
## Description
The upload of Covid samples to Gisaid is failing after the pydantic migration. The validators where not run as the default value was used.

The PR removes the non-functioning validators and replaces them with a model_validator which sets the fields.

### Fixed

- GisaidSample validation is now done via model_validator instead of field_validator
- GisaidAccession validation is now done via model_validator instead of field_validator

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Review by ChrOertlin — APPROVED** [review]
Looks good.

File: cg/meta/upload/gisaid/models.py [hunk 1]
```diff
@@ -58,22 +64,36 @@ class GisaidSample(BaseModel):
...
-    def parse_virus_name(cls, _, info: FieldValidationInfo):
-        sample_name = info.data.get("covv_subm_sample_id")
-        date = info.data.get("covv_collection_date")
-        datetime_date = datetime.strptime(date, "%Y-%m-%d")
-        return f"hCoV-19/Sweden/{sample_name}/{datetime_date.year}"
+def _generate_covv_location(region):
+    return f"Europe/Sweden/{region}"
+
+
+def _generate_covv_subm_sample_id(subm_sample_id, region_code):
```
Comment (diitaz93) [inline cg/meta/upload/gisaid/models.py]: maybe add typed args and return type hints

File: cg/meta/upload/gisaid/models.py [hunk 2]
```diff
@@ -58,22 +64,36 @@ class GisaidSample(BaseModel):
...
+    return f"{region_code}_SE100_{subm_sample_id}"
+
+
+def _generate_covv_virus_name(covv_subm_sample_id, covv_collection_date):
```
Comment (diitaz93) [inline cg/meta/upload/gisaid/models.py]: here typed args and return type hints too

**Review by diitaz93 — APPROVED** [review]
Great 👍

File: cg/meta/upload/gisaid/models.py (hunk 1 above)
Comment (Vince-janv) [reply cg/meta/upload/gisaid/models.py]: Fixed

File: cg/meta/upload/gisaid/models.py (hunk 2 above)
Comment (Vince-janv) [reply cg/meta/upload/gisaid/models.py]: Fixed

**Pull request merged**